evm/src/cpu/kernel/interpreter.rs

```diff
@@ -65,6 +65,32 @@ pub fn run_interpreter(
     )
 }
 
+pub struct InterpreterSetup {
+    pub label: String,
+    pub stack: Vec<U256>,
+    pub segment: Segment,
+    pub memory: Vec<(usize, Vec<U256>)>,
+}
+
+impl InterpreterSetup {
```

Comment:
Wow this is a really nice interface! Minor nitpick, but since it actually runs the interpreter, the name "interpreter _setup_" doesn't seem to quite capture it; maybe `InterpreterTestWrapper` or something?

Reply:
how about this new API (based on a suggestion from Daniel)?

Reply:
Sorry what are the differences? I can't tell

Reply:
Oh nvm, I can see it. It looks great!